pragma/publisher/fetchers/avnu.py

```diff
@@ -207,8 +192,7 @@ async def _fetch_decimals(self, address: str) -> int:
         )
 
         # Pass the created call to Client.call_contract
-        [decimals] = await pragma_client.client.call_contract(call)
-
+        [decimals] = await self._pragma_client().client.call_contract(call)
```

Comment:
Extracted out the `PragmaClient` to a private function in the class so that it's easier to test.